### PR TITLE
get images given in dataset.yaml

### PR DESCRIPTION
**Pull request by akmaralAW**
Checked with physicall_all/random_flight_ng/. It has two .yaml files - for all and for filtered. With this script I copy image files given in filtered.yaml, and create separate output folder.

### CODE REVIEW

File: dataset_pack.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import argparse
+import yaml
+from pathlib import Path
+import os
+import shutil
+import cv2
+
+def clean(dataset_yaml_in, output_folder):
+    robot_names = [str(f).split('/')[-1] for f in Path(dataset_yaml_in).parent.iterdir() if f.is_dir()]
```
Comment (whoenig) [inline dataset_pack.py]: not needed - the goal is to have a flat hierarchy, i.e., no subfolders in output folder

File: dataset_pack.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    output_folder = Path(output_folder)
+    if output_folder.exists() and output_folder.is_dir():
+        shutil.rmtree(output_folder)
```
Comment (whoenig) [inline dataset_pack.py]: please don't use that - it's dangerous. Just reporting an error is fine, if the output folder exists.

File: dataset_pack.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        os.mkdir(output_folder)
+
+    for robot in robot_names:
+        folder = output_folder / robot
+        os.mkdir(folder)
```
Comment (whoenig) [inline dataset_pack.py]: not needed

File: dataset_pack.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        #     dataset_yaml_out['calibration'][calib_name_new] = dataset['calibration'][calib_name]
+        #     calib_name_dict[calib_name] = calib_name_new
+    for image_name in dataset['images']:
+            # dataset_yaml_out['images'][image_name] = dataset['images'][image_name]
+            image = cv2.imread(str((Path(dataset_yaml_in).parent / image_name)))
+            if (output_folder / Path(image_name).parent).exists():
+                 cv2.imwrite(str(Path(output_folder) / image_name), image)
+            else:
+                 os.makedirs(output_folder / Path(image_name).parent)
+                 cv2.imwrite(str(Path(output_folder) / image_name), image)
```
Comment (whoenig) [inline dataset_pack.py]: no need to read/write the image using opencv. Just copy the file instead.

File: dataset_pack.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+        #     dataset_yaml_out['calibration'][calib_name_new] = dataset['calibration'][calib_name]
+        #     calib_name_dict[calib_name] = calib_name_new
+    for image_name in dataset['images']:
+            # dataset_yaml_out['images'][image_name] = dataset['images'][image_name]
+            image = cv2.imread(str((Path(dataset_yaml_in).parent / image_name)))
+            if (output_folder / Path(image_name).parent).exists():
+                 cv2.imwrite(str(Path(output_folder) / image_name), image)
+            else:
+                 os.makedirs(output_folder / Path(image_name).parent)
```
Comment (whoenig) [inline dataset_pack.py]: ideally don't create subfolders and just put the file in the output folder. If a file with that name already exists, append a number to the name, until no such file exists anymore.

File: dataset_pack.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                 cv2.imwrite(str(Path(output_folder) / image_name), image)
+            
+    # yaml.Dumper.ignore_aliases = lambda *args : True
+    # with open(Path(output_folder) / 'dataset.yaml', 'w') as outfile:
+    #         yaml.dump(dataset_yaml_out, outfile)
+    shutil.copy(dataset_yaml_in, output_folder / 'dataset.yaml')
```
Comment (whoenig) [inline dataset_pack.py]: if you change the filenames as suggested above, then you would need to use yaml.dump here instead to adjust them in the output dataset.yaml file.

**Review by whoenig — CHANGES_REQUESTED** [review]
I think this does what we are looking for. I suggest changing the output to a "flat" hierarchy without subfolders, though. This will keep the dataset.yaml file smaller and also removes some of the uglyness on how we collected data. Details, see in the comments

**Comment by akmaralAW** [comment]
All provided comments are followed. Tested with random_flights_ng dataset.

**Review by whoenig — APPROVED** [review]
Looks great - thanks!

**Pull request merged**